subprojects/platform-base/src/main/java/org/gradle/api/internal/resolve/DefaultProjectModelResolver.java

```diff
@@ -36,9 +36,11 @@ public ModelRegistry resolveProjectModel(String path) {
         }
 
         // TODO This is a brain-dead way to ensure that the reference project's model is ready to access
-        projectInternal.prepareForRuleBasedPlugins();
-        projectInternal.evaluate();
-        projectInternal.getTasks().discoverTasks();
-        return projectInternal.getModelRegistry();
+        return projectInternal.getMutationState().withMutableState(() -> {
+            projectInternal.prepareForRuleBasedPlugins();
+            projectInternal.evaluate();
+            projectInternal.getTasks().discoverTasks();
+            return projectInternal.getModelRegistry();
+        });
```

Comment:
Hmm, it looks like there might be a deadlock/livelock problem in

NativeDependentComponentsReportIntegrationTest.can show dependent components in parallel

https://builds.gradle.org/viewLog.html?buildId=29515919&buildTypeId=Gradle_Check_Quick_1_platformBase&tab=testsInfo&branch_Gradle_Check_Quick_1=sg%2F61%2Ftypo-fix

This may be harder to fix than I originally thought.

Reply:
Maybe we could scope the fix to the very narrow case of pre-built binaries. I will have a quick look at this.